nucleus/glassfish-jul-extension/src/test/java/org/glassfish/main/jul/GlassFishLogManagerLifeCycleTest.java

```diff
@@ -133,6 +136,7 @@ public void startReconfiguration() throws Exception {
             () -> assertEquals(GlassFishLoggingStatus.CONFIGURING, MANAGER.getLoggingStatus()),
             () -> assertNull(COLLECTOR.pop(), "COLLECTOR should be empty after reconfiguration started"));
         doLog(Level.INFO, "Log after reconfiguration started", 0);
+        Thread.sleep(10L);
```

Comment:
In all these tests should not be any sleeps, they make these tests ineffective. Also the line before has 0 sleep time, while the thread started before the sleep (but still could be executed before or after it).
I have to think what kind of error I intended to catch by this.

Reply:
I added this sleep to ensure that the logging actually happens before the assertion on the next line. Otherwise the assertion doesn't have any value - it could be asserted before the log happens and then the test passes.

Should I remove the sleep here?

Reply:
Could `doLog` return something that the code here could check if it's safe to proceed?

Reply:
I have no idea. Now it looks to me that @dmatej  intended to do the check immediately. It’s supposed to test locking, so maybe all should run as soon as possible. Although it would be good to clarify what the tests intends to verify.

Reply:
Do you mean the executor I created here in rhe LogFileManager? It creates a thread only when it’s needed and disposes it after 100ms if it’s inactive. The only thing that is kept all the time is the executor itself, with no threads most of the time.

A better approach would be as you suggest - it would be enough to create a single threaded executor in the scope of roll method and use it for all async logging and then shut it down at the end of the roll method.